www-src/app/lib/android_picture_folder_handler.coffee

```diff
@@ -0,0 +1,67 @@
+DesignDocuments = require "../replicator/design_documents"
+log = require("./persistent_log")
+    prefix: "AndroidPictureFolderHandler"
+    date: true
+
+
+module.exports = class AndroidPictureFolderHandler
+
+
+    ERROR_FOLDER_EXIST: "Pictures folder isn't created."
+
+
+    constructor: (@deviceName, @db) ->
+        @deviceName ?= app.replicator.config.deviceName
+        @db ?= app.replicator.db
+
+
+    getOrCreate: (callback) ->
+        log.info "getOrCreate"
+
+        @get (err, folderPictures) =>
+            if err
+                if err.message is @ERROR_FOLDER_EXIST
+                    return @create callback
+
+                return callback err
+
+            callback null, folderPictures
+
+
+    get: (callback) ->
+        log.info "get"
+
+        options =
+            key: [
+                ""                               # path
+                "1_#{t('photos').toLowerCase()}" # emit in design doc for folder
```

Comment:
I'm not a big fan of using a translated string as a key for searching in pouchdb (for example, if the user change its locale, the folder won't be found). I think it should be better to separate the showing concern from the searching concern. But it can wait. It's more something to keep in head in the long term.